### PR TITLE
Update relayer guide

### PR DESCRIPTION
**Pull request by CallumGrindle**
Updates relayer API guide to reflect smart contract version change.

Adds guide on how to use executeRelayCall on the client and server of a transaction relayer service

### CODE REVIEW

File: docs/guides/key-manager/execute-relay-call.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+sidebar_label: 'Execute Relay Call'
+sidebar_position: 2
+---
+
+# Execute Relay Call
+
+The [LSP6 KeyManager standard](../../standards/universal-profile/lsp6-key-manager.md) enables anybody to execute a transation on behalf of a Universal Profile, given they have a valid transaction which has been signed by a key which controls the Universal Profile.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:8]: Is there already a set convention for LSP terms @CJ42 ? Otherwise I would suggest to either write `LSP6 KeyManager Standard` , or keep the word `standard` lowercase but out of the link.

Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:8]:
```suggestion
The [LSP6 KeyManager standard](../../standards/universal-profile/lsp6-key-manager.md) enables anybody to execute a transaction on behalf of a Universal Profile, given they have a valid transaction signed by a key that controls the Universal Profile.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+This enables use cases such as Transaction Relayer Services to be possible where users can send their transaction details to a third party to be executed, moving the gas cost burden away from the user who owns the Universal Profile.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:10]:
```suggestion
Relayed execution enables use cases such as Transaction Relayer Services to be possible where users can send their transaction details to a third party to be executed, moving the gas cost burden away from the user who owns the Universal Profile.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+For example, Alice who owns a Universal Profile can send an encoded transaction which updates her [LSP3Profile](../../standards/universal-profile/lsp3-universal-profile-metadata.md) picture to a second user Bob, who executes the transaction and pays the gas cost of the transaction on behalf of Alice.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:12]:
```suggestion
For example, Alice, who owns a Universal Profile, can send an encoded transaction that updates her [LSP3Profile](../../standards/universal-profile/lsp3-universal-profile-metadata.md) picture to a second user Bob, who executes the transaction and pays the gas cost of the transaction on behalf of Alice.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+For this, Bob needs to know the encoded transaction ABI to be executed, the signature and the nonce of the key which signed the transaction.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:14]: IMO, its a bit too wordy and hard to grasp

```suggestion
To execute such a transaction, Bob needs to know: 
- the encoded ABI of the transaction that will get executed, 
- the transactions signature, and
- the nonce of the key which signed the transaction.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+The transaction is then executed via the [LSP6KeyManager](../../standards/universal-profile/lsp6-key-manager.md) function `executeRelayCall`.
+
+## Generate the Signed Transaction Payload
+
+:::info
+This example shows how to prepare a transaction to be executed by a third party. This logic can be implemented client side and then sent to a third party application or service such as a Transaction Relay service to be executed.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:21]:
```suggestion
This example shows how to prepare a transaction to be executed by a third party. This logic can be implemented client-side and then sent to a third-party application or service such as a Transaction Relay service to be executed.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+:::
+
+To encode a transaction we need the address of the Universal Profile smart contract and the private key of a controller key which has sufficient [LSP6 permissions](../../standards/universal-profile/lsp6-key-manager.md#permissions) to execute the transaction.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:24]:
```suggestion
To encode a transaction, we need the address of the Universal Profile smart contract and the private key of a controller key with sufficient [LSP6 permissions](../../standards/universal-profile/lsp6-key-manager.md#permissions) to execute the transaction.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+```typescript
+const controllingAccountPrivateKey = '0x...';
+const myUpAddress = '0x...';
+```
+
+Get the `nonce` of the controller key from the KeyManager by instantiating the KeyManager Contract instance and calling `getNonce`.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:31]:
```suggestion
Get the `nonce` of the controller key from the KeyManager by instantiating the KeyManager smart contract instance and calling its `getNonce` function.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+const controllerAccount =
+  web3.eth.accounts.privateKeyToAccount(controllerPrivateKey);
+const channelId = 0;
+
+const nonce = await KeyManager.methods
+  .getNonce(controllerAccount.address, channelId)
+  .call();
+```
+
+Encode the ABI of the transaction you want to be executed. In this case a LYX transfer to a recipient address.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:59]:
```suggestion
Encode the ABI of the transaction you want to be executed. In this case, an LYX transfer to a recipient address.
```

We use vocal terms in `an NFT` as well, so I would also use them for the term `LYX`

File: docs/guides/key-manager/execute-relay-call.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+Now the `signature`, `abiPayload`, `nonce` and `keyManagerAddress` can be sent to a third party to execute the transaction using `executeRelayCall`.
+
+## Execute via `executeRelayCall`
+
+:::info
+This example shows how a third party can execute a transaction on behalf of another user.
+:::
+
+To execute a signed transaction ABI payload requires the **KeyManager contract address**, the **transaction abi payload**, **signed transaction payload** and **nonce** of the controller key which signed the transaction.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:94]:
```suggestion
To execute a signed transaction ABI payload requires the **KeyManager contract address**, the **transaction ABI payload**, **signed transaction payload**, and **nonce** of the controller key which signed the transaction.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+:::note
+To get the KeyManager address from the UniversalProfile address call the `owner` function on the Universal Profile contract.
```
Comment (fhildeb) [inline docs/guides/key-manager/execute-relay-call.md:97]:
```suggestion
To get the KeyManager address from the UniversalProfile address, call the `owner` function on the Universal Profile contract.
```

**Review by fhildeb — CHANGES_REQUESTED** [review]
Great job, more relayer information is is urgently needed! Just improved grammar and simplified some sentences.

**Review by fhildeb — APPROVED** [review]

File: docs/guides/key-manager/execute-relay-call.md (hunk 1 above)
Comment (CJ42) [inline docs/guides/key-manager/execute-relay-call.md]: _Suggested changes._

I think also maybe would be good to link to the **permissions** section on the LSP6 Standard page for the word "permissions" at the end (I cannot find the correct link).


```suggestion
The [LSP6 KeyManager standard](../../standards/universal-profile/lsp6-key-manager.md) enables any `address` (EOA or contract) to execute a transaction on behalf of a Universal Profile, given they have a valid signature for a transaction and have the right permissions on the Universal Profile.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+The [LSP6 KeyManager standard](../../standards/universal-profile/lsp6-key-manager.md) enables anybody to execute a transation on behalf of a Universal Profile, given they have a valid transaction which has been signed by a key which controls the Universal Profile.
+
+This enables use cases such as Transaction Relayer Services to be possible where users can send their transaction details to a third party to be executed, moving the gas cost burden away from the user who owns the Universal Profile.
+
+For example, Alice who owns a Universal Profile can send an encoded transaction which updates her [LSP3Profile](../../standards/universal-profile/lsp3-universal-profile-metadata.md) picture to a second user Bob, who executes the transaction and pays the gas cost of the transaction on behalf of Alice.
+
+For this, Bob needs to know the encoded transaction ABI to be executed, the signature and the nonce of the key which signed the transaction.
+
+The transaction is then executed via the [LSP6KeyManager](../../standards/universal-profile/lsp6-key-manager.md) function `executeRelayCall`.
```
Comment (CJ42) [inline docs/guides/key-manager/execute-relay-call.md:20]: Maybe we could add a link to `executeRelayCall(...)` to the contract API docs?

File: docs/guides/key-manager/execute-relay-call.md (hunk 2 above)
Comment (CJ42) [inline docs/guides/key-manager/execute-relay-call.md]:
```suggestion
This enables use cases such as Transaction Relay Services to be possible where users can send their transaction details to a third party to be executed, moving the gas cost burden away from the user who owns the Universal Profile.
```

File: docs/guides/key-manager/execute-relay-call.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+To encode a transaction we need the address of the Universal Profile smart contract and the private key of a controller key which has sufficient [LSP6 permissions](../../standards/universal-profile/lsp6-key-manager.md#permissions) to execute the transaction.
+
+```typescript
+const controllingAccountPrivateKey = '0x...';
+const myUpAddress = '0x...';
+```
+
+Get the `nonce` of the controller key from the KeyManager by instantiating the KeyManager Contract instance and calling `getNonce`.
+
+The `channelId` is used to prevent nonce conflicts when multiple apps send transactions to the same KeyManager at the same time. Read more about [out of order execution here](../../standards/universal-profile/lsp6-key-manager.md#out-of-order-execution).
```
Comment (CJ42) [inline docs/guides/key-manager/execute-relay-call.md:37]: These two sentences describe really well.

The example of multi-channel nonce in particular is awesome! (never thought about this use case @YamenMerhi ).

🚀 🎩

File: docs/guides/key-manager/execute-relay-call.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+  myUpAddress,
+);
+
+const keyManagerAddress = await myUniversalProfile.methods.owner().call();
+const KeyManager = new web3.eth.Contract(
+  KeyManagerContract.abi,
+  keyManagerAddress,
+);
+
+const controllerAccount =
```
Comment (CJ42) [inline docs/guides/key-manager/execute-relay-call.md:54]: can this be in just one line?

File: docs/guides/key-manager/execute-relay-call.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+```typescript title="Encode transaction ABI"
+const abiPayload = myUniversalProfile.methods.execute(
+    0,
+    '0x...', // Recipient address
+    web3.utils.toWei('1'),
+    '0x'
+).encodeABI()) ;
+```
+
+Afterward, sign the transaction message from the controller key of the Universal Profile.
```
Comment (CJ42) [inline docs/guides/key-manager/execute-relay-call.md:74]:
```suggestion
Afterwards, sign the transaction message from the controller key of the Universal Profile.
```

**Review by CJ42 — COMMENTED** [review]
Awesome guide!
